### PR TITLE
swarm/network/stream: use bzzAddr for proximity calculation

### PR DESCRIPTION
**Pull request by holisticode**
This PR uses `BzzAddr` instead of `enode.ID` to use `chunk.Proximity()` for `po` calculations between nodes.

This PR currently passes the tests but it is important to understand that it produces errors which do NOT result in the tests failing....check the logs!

### CODE REVIEW

File: swarm/network/stream/peer.go [hunk 1]
```diff
@@ -58,6 +58,7 @@ var ErrMaxPeerServers = errors.New("max peer servers")
 // Peer is the Peer extension for the streaming protocol
 type Peer struct {
 	*protocols.Peer
+	bzzPeer  *network.BzzPeer
```
Comment (nonsense) [inline swarm/network/stream/peer.go:59]: The `bzzPeer` has access to the `protocols.Peer`, so maybe we don't need the embedding on line 60 ?

File: swarm/network/stream/peer.go [hunk 2]
```diff
@@ -77,9 +78,10 @@ type WrappedPriorityMsg struct {
 }
 
 // NewPeer is the constructor for Peer
-func NewPeer(peer *protocols.Peer, streamer *Registry) *Peer {
+func NewPeer(peer *network.BzzPeer, streamer *Registry) *Peer {
 	p := &Peer{
-		Peer:         peer,
+		Peer:         peer.Peer,
```
Comment (nonsense) [inline swarm/network/stream/peer.go]: Explicitly keeping a pointer to peer.Peer seems redundant.

File: swarm/network/stream/peer.go (hunk 1 above)
Comment (holisticode) [reply swarm/network/stream/peer.go:59]: I applied a change to address this which resulted in quite a lot of cascading changes...I hope this does not introduce some new level of "ugliness"?

File: swarm/network/stream/peer.go (hunk 2 above)
Comment (holisticode) [reply swarm/network/stream/peer.go]: See above

File: swarm/network/stream/peer.go (hunk 1 above)
Comment (nonsense) [reply swarm/network/stream/peer.go:59]: @holisticode all the `peer` structs are confusing to me. I suggested the change in case it didn't result in many cascading changes. If this is not the case, feel free to just address this particular problem, and once this is correct, we can review and discuss how to refactor :)

File: swarm/network/stream/delivery.go [hunk 3]
```diff
@@ -80,7 +80,7 @@ func (d *Delivery) handleRetrieveRequestMsg(ctx context.Context, sp *Peer, req *
 
 	var cancel func()
 	// TODO: do something with this hardcoded timeout, maybe use TTL in the future
-	ctx = context.WithValue(ctx, "peer", sp.ID().String())
+	ctx = context.WithValue(ctx, "peer", sp.bzzPeer.ID().String())
```
Comment (nonsense) [inline swarm/network/stream/delivery.go:83]: This is changing behaviour on `master`. One of the things that I am addressing in `simple-fetchers` is the use of `context` as a container for values that change the logic of execution.

I am not sure if it is okay to change `enode.ID` to `bzzAddr` here.

File: swarm/network/stream/delivery.go [hunk 4]
```diff
@@ -177,7 +177,7 @@ func (d *Delivery) handleChunkDeliveryMsg(ctx context.Context, sp *Peer, req int
 				// we removed this log because it spams the logs
 				// TODO: Enable this log line
 				// log.Warn("invalid chunk delivered", "peer", sp.ID(), "chunk", msg.Addr, )
-				msg.peer.Drop()
+				msg.peer.bzzPeer.Drop()
```
Comment (nonsense) [inline swarm/network/stream/delivery.go:180]: Why do we have to change this? I think you are changing way too many places from peer to bzzPeer, which is essentially the same.

File: swarm/network/stream/delivery.go [hunk 5]
```diff
@@ -164,7 +164,7 @@ func (d *Delivery) handleChunkDeliveryMsg(ctx context.Context, sp *Peer, req int
 		mode = chunk.ModePutSync
 	}
 
-	log.Trace("handle.chunk.delivery", "ref", msg.Addr, "from peer", sp.ID())
+	log.Trace("handle.chunk.delivery", "ref", msg.Addr, "from peer", sp.bzzPeer.ID())
```
Comment (nonsense) [inline swarm/network/stream/delivery.go:167]: Why is this necessary now, if we are not going to be refactoring all the peers in this PR?

File: swarm/network/stream/messages.go [hunk 6]
```diff
@@ -75,13 +75,13 @@ type RequestSubscriptionMsg struct {
...
-	log.Debug(fmt.Sprintf("handleRequestSubscription: streamer %s to subscribe to %s with stream %s", p.streamer.addr, p.ID(), req.Stream))
-	if err = p.streamer.Subscribe(p.ID(), req.Stream, req.History, req.Priority); err != nil {
+	log.Debug(fmt.Sprintf("handleRequestSubscription: streamer %s to subscribe to %s with stream %s", p.streamer.addr, p.bzzPeer.ID(), req.Stream))
+	if err = p.streamer.Subscribe(p.bzzPeer.ID(), req.Stream, req.History, req.Priority); err != nil {
 		// The error will be sent as a subscribe error message
 		// and will not be returned as it will prevent any new message
 		// exchange between peers over p2p. Instead, error will be returned
 		// only if there is one from sending subscribe error message.
-		err = p.Send(ctx, SubscribeErrorMsg{
+		err = p.bzzPeer.Send(ctx, SubscribeErrorMsg{
```
Comment (nonsense) [inline swarm/network/stream/messages.go:84]: Why do we need to change this as well? Whether we call `Send` on `p` or `bzzPeer` is the same, no?

File: swarm/network/stream/peer.go (hunk 1 above)
Comment (nonsense) [reply swarm/network/stream/peer.go:59]: @holisticode I am sorry if you have understood my comment as a request to do that change - it was not intended as such.

I think the new cascading change is much worse to what we have right now, and will also result in many conflicts with other branches we have in progress right now.

Furthermore this cascading change is not addressing the underlying problem - which is the mess of all the different `peer` structs and their inheritance.

---

I suggest we just fix the proximity calculations if possible, and not do such cascading change throughout the codebase, unless we actually address the underlying problem of the design.

**Comment by holisticode** [comment]
Obsolete due to https://github.com/ethersphere/go-ethereum/pull/1336
